pages/k8s/charm-kubernetes-worker.md

```diff
@@ -265,6 +268,77 @@ the setting of conntrack-max-per-core vs nf_conntrack_max.
 <!-- CONFIG ENDS -->
 
 
+<a id="k8s-services"> </a>
+## Configuring K8s services
+
+### IPVS (IP Virtual Server)
+
+This requires configuration of both the `kubernetes-master` and
+`kubernetes-worker` charms. Please see the configuration section on
+the [kubernetes-master page](../charm-kubernetes-master#config-ipvs).
+
+### HugePages
+
+HugePages are a standard memory management feature of the Linux kernel to
+decrease overhead for processes which consume large amounts of memory.
+
+Kubernetes includes support for using HugePages with pods (see the
+[upstream documentation](https://kubernetes.io/docs/tasks/manage-hugepages/scheduling-hugepages/)).
+
+To use HugePages in your pods with **Charmed Kubernetes**, it is necessary to
+update the configuration for the workers:
+
+1.  Fetch the current 'sysctl' configuration from the worker:
+    ```bash
+    juju config kubernetes-worker sysctl
+    ```
+    This should return a sting of config options, e.g.:
+    ```
+    { net.ipv4.conf.all.forwarding : 1, net.ipv4.neigh.default.gc_thresh1 : 128, net.ipv4.neigh.default.gc_thresh2 : 28672, net.ipv4.neigh.default.gc_thresh3 : 32768, net.ipv6.neigh.default.gc_thresh1 : 128, net.ipv6.neigh.default.gc_thresh2 : 28672, net.ipv6.neigh.default.gc_thresh3 : 32768, fs.inotify.max_user_instances : 8192, fs.inotify.max_user_watches: 1048576 }
+    ```
+2.  The config option for HugePages is `vm.nr_hugepages`. To add this
+    configuration, you should **append** it to the string and set the whole
+    configuration. For example, for 100 2Mi pages:
+    ```bash
+    juju config kubernetes-worker sysctl="{ net.ipv4.conf.all.forwarding : 1, net.ipv4.neigh.default.gc_thresh1 : 128, net.ipv4.neigh.default.gc_thresh2 : 28672, net.ipv4.neigh.default.gc_thresh3 : 32768, net.ipv6.neigh.default.gc_thresh1 : 128, net.ipv6.neigh.default.gc_thresh2 : 28672, net.ipv6.neigh.default.gc_thresh3 : 32768, fs.inotify.max_user_instances : 8192, fs.inotify.max_user_watches: 1048576, vm.nr_hugepages: 100}"
+    ```
+3.  HugePages can now be consumed via container level resource requirements
+    using the resource name <code>hugepages-&lt;size&gt;</code>.
+
+    For example:
+
+    ```yaml
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: hugepages-test
+    spec:
+      containers:
+      - image: ubuntu:latest
+        command:
+        - sleep
+        - inf
+        name: example
+        volumeMounts:
+        - mountPath: /hugepages
+          name: hugepage
+        resources:
+          limits:
+            hugepages-2Mi: 100Mi
+            memory: 100Mi
+          requests:
+            memory: 100Mi      
+      volumes:
+      - name: hugepage
+        emptyDir:
+          medium: HugePages  
+    ```
+    &nbsp;
```

Comment:
In the rendered output this appears to add an additional unnecessary blank line.

Reply:
the intentional unnecessary blank line is the one amazing trick I discovered to make the preceding YAML block to render correctly, after literally hours of research into code blocks in numbered indented lists.
it doesn't have to be blank though, perhaps I can add a final comment instead to make it look better.